mongomock/aggregate.py

```diff
@@ -256,6 +259,40 @@ def _handle_comparison_operator(self, operator, values):
             'aggregation pipeline, it is currently not implemented '
             ' in Mongomock.' % operator)
 
+    def _handle_string_operator(self, operator, values):
+        if operator == '$toLower':
+            parsed = self.parse(values)
+            return str(parsed).lower() if parsed is not None else ''
+        if operator == '$toUpper':
+            parsed = self.parse(values)
+            return str(parsed).upper() if parsed is not None else ''
+        if operator == '$concat':
+            parsed_list = [self.parse(value) for value in values]
+            return None if None in parsed_list else ''.join([str(x) for x in parsed_list])
+        if operator == '$substr':
+            if len(values) != 3:
+                raise OperationFailure('substr must have 3 items')
+            string = str(self.parse(values[0]))
+            first = self.parse(values[1])
+            length = self.parse(values[2])
+            if string is None or first < 0:
+                return ''
+            second = len(string) if length < 0 else first + length
+            return string[first:second]
+        if operator == '$strcasecmp':
+            if len(values) != 2:
+                raise OperationFailure('strcasecmp must have 2 items')
+            a, b = str(self.parse(values[0])), str(self.parse(values[1]))
```

Comment:
Add two test for stcasecmp, one for this, and one for when it succeeds and one for when it fails.

Reply:
ok